hack/dockerfiles/test.buildkit.Dockerfile

```diff
@@ -0,0 +1,236 @@
+# syntax = tonistiigi/dockerfile:runmount20180925
+
+ARG RUNC_VERSION=00dc70017d222b178a002ed30e9321b12647af2d
+ARG CONTAINERD_VERSION=v1.1.3
+# containerd v1.0 for integration tests
+ARG CONTAINERD10_VERSION=v1.0.3
+# available targets: buildkitd, buildkitd.oci_only, buildkitd.containerd_only
+ARG BUILDKIT_TARGET=buildkitd
+ARG REGISTRY_VERSION=2.6
+ARG ROOTLESSKIT_VERSION=20b0fc24b305b031a61ef1a1ca456aadafaf5e77
+
+# available targets: buildkitd, buildkitd.oci_only, buildkitd.containerd_only
+ARG BUILDKIT_TARGET=buildkitd
+
+# git stage is used for checking out remote repository sources
+FROM --platform=$BUILDPLATFORM alpine AS git
+RUN apk add --no-cache git
+
+# xgo is a helper for golang cross-compilation
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:golang AS xgo
+
+# gobuild is base stage for compiling go/cgo
+FROM --platform=$BUILDPLATFORM golang:1.11 AS gobuild-minimal
+COPY --from=xgo / /
+RUN apt-get update && apt-get install --no-install-recommends -y libseccomp-dev file
+
+# on amd64 you can also cross-compile to other platforms
+FROM gobuild-minimal AS gobuild-cross-amd64
+RUN dpkg --add-architecture s390x && \
+  dpkg --add-architecture ppc64el && \
+  dpkg --add-architecture armel && \
+  dpkg --add-architecture armhf && \
+  dpkg --add-architecture arm64 && \
+  apt-get update && \
+  apt-get install -y \
+  gcc-s390x-linux-gnu libc6-dev-s390x-cross libseccomp-dev:s390x \
+  crossbuild-essential-ppc64el libseccomp-dev:ppc64el \
+  crossbuild-essential-armel libseccomp-dev:armel \
+  crossbuild-essential-armhf libseccomp-dev:armhf \
+  crossbuild-essential-arm64 libseccomp-dev:arm64 \
+  --no-install-recommends
+
+# define all valid target configurations for compilation
+FROM gobuild-minimal AS gobuild-amd64-amd64
+FROM gobuild-minimal AS gobuild-arm-arm
+FROM gobuild-minimal AS gobuild-s390x-s390x
+FROM gobuild-minimal AS gobuild-ppc64le-ppc64le
+FROM gobuild-minimal AS gobuild-arm64-arm64
+FROM gobuild-cross-amd64 AS gobuild-amd64-arm
+FROM gobuild-cross-amd64 AS gobuild-amd64-s390x
+FROM gobuild-cross-amd64 AS gobuild-amd64-ppc64le
+FROM gobuild-cross-amd64 AS gobuild-amd64-arm64
+FROM gobuild-$BUILDARCH-$TARGETARCH AS gobuild-base
+
+# runc source
+FROM git AS runc-src
+ARG RUNC_VERSION
+WORKDIR /usr/src
+RUN git clone git://github.com/opencontainers/runc.git runc \
```

Comment:
Can we experiment `ADD git://` source op?

Reply:
Maybe `FROM git://` ?